oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/authentication/CodeVerifierAuthenticator.java

```diff
@@ -140,7 +140,8 @@ private static boolean authorizationCodeGrant(Map<String, Object> parameters) {
 		// @formatter:off
 		return AuthorizationGrantType.AUTHORIZATION_CODE.getValue().equals(
 				parameters.get(OAuth2ParameterNames.GRANT_TYPE)) &&
-				parameters.get(OAuth2ParameterNames.CODE) != null;
+				parameters.get(OAuth2ParameterNames.CODE) != null &&
```

Comment:
If the grant type is `authorization_code` and the `code` is empty, it should not return `false` and instead should error.

Try this instead:

```java
if (!AuthorizationGrantType.AUTHORIZATION_CODE.getValue().equals(parameters.get(OAuth2ParameterNames.GRANT_TYPE))) {
	return false;
}
if (!StringUtils.hasText((String) parameters.get(OAuth2ParameterNames.CODE))) {
	throwInvalidGrant(OAuth2ParameterNames.CODE);
}
return true;
```

Reply:
Please review new changes.